torch/distributed/_pipeline/sync/stream.py

```diff
@@ -104,7 +104,8 @@ def record_stream(tensor: torch.Tensor, stream: AbstractStream) -> None:
         #
         tensor = tensor.new_empty([0]).set_(tensor.storage())
 
-        tensor.record_stream(as_cuda(stream))
+        # Typechecking: torch.cuda.Stream is incompatible with torch._C.Stream
+        tensor.record_stream(as_cuda(stream))  # type: ignore[arg-type]
```

Comment:
should we change `tensor.record_stream` to return `AbstractStream` instead of `Stream` then?

Reply:
tensor.record_stream returns `None` and takes an argument of `torch._C.Stream`. This function is defined in torch._C.__init__.pyi, where we don't have the definition of `AbstractStream`. Neither can I find its related binding in C++. So I will leave it as-is until the next clean-up.